doc/macsec-gearbox/Copper_PHY_Attributes_Proposal.md

```diff
@@ -0,0 +1,308 @@
+} sai_port_mdix_mode_config_t;
+
+/**
+ * @brief MDIX mode status for the port
+ *
+ * @type sai_port_mdix_mode_status_t
+ * @flags READ_ONLY
+ */
+SAI_PORT_ATTR_MDIX_MODE_STATUS,
+
+/**
+ * @brief MDIX mode configuration for the port
+ *
+ * @type sai_port_mdix_mode_config_t
+ * @flags CREATE_AND_SET
+ * @default SAI_PORT_MDIX_MODE_CONFIG_AUTO
+ */
+SAI_PORT_ATTR_MDIX_MODE_CONFIG,
+
+```
+## Master/Slave Auto-negotiation Configuration
+
+In 1000BASE-T mode, one end of the link must be configured as the timing master and the other end as the slave. Master/slave configuration is performed by the auto-negotiation function. The auto-negotiation function first looks at the manual master/slave configuration modes advertised by the local PHY and the link partner. If both ends of the link attempt to force the same manual configuration (both master or both slave), PHY will try to link up and restart auto-negotiation if needed. SAI\_PORT\_ATTR\_AUTO\_NEG\_CONFIG\_MODE can be used to set the PHY either in Master, Slave, Auto mode (defer to hardware) or disable auto-negotiation mode.
+
+```
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_AUTO_NEG_CONFIG_MODE
+ * Used for auto negotiation mode to configure master or slave mode
+ */
+typedef enum _sai_port_auto_neg_config_mode_t
+{
+    /** Auto neg configuration mode disabled */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_DISABLED,
+
+    /** Auto neg mode auto */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_AUTO,
+
+    /** Auto neg mode slave */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_SLAVE,
+
+    /** Auto neg mode master */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_MASTER
+} sai_port_auto_neg_config_mode_t;
+
+/**
+ * @brief Configure auto negotiation configuration mode for the port
+ *
+ * @type sai_port_auto_neg_config_mode_t
+ * @flags CREATE_AND_SET
+ * @default SAI_PORT_AUTO_NEG_CONFIG_MODE_DISABLED
+ * @validonly SAI_PORT_ATTR_AUTO_NEG_MODE == true
+ */
+SAI_PORT_ATTR_AUTO_NEG_CONFIG_MODE,
+```
+## 1000BaseX and SGMII Slave Autodetect
+
+When set to true, PHY device can automatically switch between 1000BaseX and SGMII-Slave mode and 1000Base-X or SGMII-Slave module can get a link up without modifying PHY configurations. This attribute is only valid for Fiber interfaces and when the speeds are 1G or 100M.
+
+```
+/**
+ * @brief Enable auto detection between 1000X and SGMII slave mode
+ *
+ * @type bool
+ * @flags CREATE_AND_SET
+ * @default false
+ * @validonly SAI_PORT_ATTR_MEDIA_TYPE == SAI_PORT_MEDIA_TYPE_FIBER and SAI_PORT_ATTR_SPEED == 1000
+ */
+SAI_PORT_ATTR_1000X_SGMII_SLAVE_AUTODETECT,
+
+```
+
+## EEE Latency Mode
+
+EEE Latency mode allows user to configure a fixed or variable latency when EEE is enabled. Variable latency can be configured using SAI_PORT_ATTR_EEE_WAKE_TIME
+
+```
+/**
+ * @brief Configure EEE latency mode
+ *
+ * False: Fixed latency
+ * True: Variable latency
+ *
+ * @type bool
+ * @flags CREATE_AND_SET
+ * @default false
+ */
+SAI_PORT_ATTR_EEE_LATENCY_MODE,
```

Comment:
there is already SAI_PORT_ATTR_EEE_ENABLE attribute. How is this new attribute related to the EEE_ENABLED attribute?

Reply:
I need to update the documentation. This attribute has been changed to SAI_PORT_ATTR_EEE_LATENCY_VARIABLE. Its updated in saiport.h. Some PHYs support variable latency mode. This attribute will be used to enable variable latency mode vs fixed mode. SAI_PORT_ATTR_EEE_ENABLE will be used to enable/disable EEE.

Reply:
Will add a condition to check SAI_PORT_ATTR_EEE_ENABLE